### ISSUE TITLE
Are system/audit date fields supported (CreatedDate,LastModifiedDate,etc.) ?

### ISSUE THREAD

**Issue by pgonzaleznetwork** [RetekBacsi/TestDataLoader#3]
As per the example on #2 , my test involves setting the CreatedDate of 44 case records. This is not optional as our code will execute a different logic based on the case created date, and we need to be able to test for this condition.

When using the `Test.loadData()` standard method, we are able to specify the CreatedDate of a record. If not using the loadData() method, we can always specify the date using the `setCreatedDate()` method: https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_methods_system_test.htm#apex_System_Test_setCreatedDate

I don't see this method being used anywhere in the code of this library, so I'm assuming this is not supported?

Could you add support for this? This would again bring this library to feature parity with the standard mechanism for using static resources for test data.

**Comment by RetekBacsi** [comment]
Hi There,

Please try the current version, it has support for CreatedDate. That's the only system/audit field currently supported.

**Comment by pgonzaleznetwork** [comment]
Thanks for working on this. This doesn't seem to work when I also specify the CloseDate (this is a Case record)

```
<CreatedDate>2021-05-01T10:00:00.000+0000</CreatedDate>
<ClosedDate>2021-05-24T10:00:00.000+0000</ClosedDate>
```

I get the error 

`18:42:48:885 FATAL_ERROR System.DmlException: Insert failed. First exception on row 37; first error: FIELD_INTEGRITY_EXCEPTION, Closed Date(Mon May 24 10:00:00 GMT 2021) before Create Date(Sat Nov 13 18:42:41 GMT 2021).: [ClosedDate]`

You'll see that I'm setting the close date to be 24 days after the created date. However, the error states that the created date is today (Nov 13 at the time of this writing), which is incorrect, the created date should be what I specified in the CreatedDate tag.

I think you need to change the order of operations here. First set the CreatedDate, and then the CloseDate.

**Comment by RetekBacsi** [comment]
Unfortunately that looks like a Salesforce limitation. CreatedDate is not writable, however it can be set after creation using the setCreatedDate method you linked above.
The ClosedDate is writable, but you can't set it to a value before CreatedDate. I need to tweak the logic a bit, to enable deferred field updates for normal fields too...

**Comment by pgonzaleznetwork** [comment]
Yeah, I understand. Unfortunately without this I cannot implement this in my org and I'm changing jobs this Monday so I won't be able to test it anymore for a real scenario. I'd love to test it though in a dev org if/once you push a fix for this. I liked editing the xml file much more than the csv one, so I really want to use this library! :)
